src/3rd_party_config.c

```diff
@@ -0,0 +1,296 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "3rd_party_internal.h"
+#include "config.h"
+#include "lib/config_parser.h"
+#include "lib/list.h"
+#include "lib/sys.h"
+#include "swupd.h"
+#include <errno.h>
+#include <stdlib.h>
+
+#ifdef THIRDPARTY
+
+#define REPO_CONFIG_DIR "3rd_party"
+#define REPO_CONFIG_CONFIG_FILE "repo.ini"
+#define MAX_REPO_ALLOWED_FIELDS 2
+#define MAX_GLOBAL_ALLOWED_FIELDS 2
+#define MAX_FILE_PATH 1000
+
+/** @brief Storing global key,value pairs. */
+struct global {
+	void *value1;
+	void *value2;
+};
+
+/** @brief Internal Memory representation of a repo.  */
+struct repo {
+	char *repo_name;
+	char *url;
+};
+
+/** @brief List of repos. */
+struct list *repos_kv;
+
+/** @brief We populate this once every swupd init and use it many times. */
+static char *repo_config_file_path;
+
+/**
+ * @brief This function is called by the repo ini parser It is called
+ * once for every line which is successfully parsed with a key,value,section
+ * Once a section is completely read, it is stored in the global lists.
+ *
+ * @param section A string containing section of the file
+ * @param key A string containing key
+ * @param value A string containing value
+ *
+ * @returns True for every match of section with REPO or GLOBAL, false otherwise
+ */
+bool parse_key_values(char *section, char *key, char *value, void UNUSED_PARAM *data)
+{
+	static int repo_field_count = 0;
+	static int global_field_count = 0;
+	bool ret = false;
+
+	if (!strncmp(section, "REPO", strlen(section))) {
```

Comment:
So I will try to answer the below questions all at once:

So I envision the repo config file looks more like

```
[REPO]
repo-name=repo_foo1
repo-URL=http://xys1.com
repo-version=1003

[REPO]
repo-name=repo_foo2
repo-URL=http://xys2.com
repo-version=1003
```

The reason why I took this approach is 
a.) IF we have pre-defined sections like GLOBAL, REPO, BUNDLE, then it gives more control to dipcher what a given key-value makes sense for that section.

For eg: in future we can reject something like repo_description="xyz" because we know that field does not really exist for a "[REPO]" section.   

b.) Pre-pending repo with helps use prevent name conflicts in future. Like name for some other section. So I see it more as a benefit ?

Reply:
I don't think you can have the same section multiple times in an INI file.

Reply:
So I think I misunderstood a little..how ini file is represented. Your understanding is consistent with Otavio's. Changing this and addressing your concerns in next fix.